### ISSUE TITLE
Discriminator predicts only one class for all samples

### ISSUE THREAD

**Issue by OsmanMutlu** [OsmanMutlu/Pytorch-GANBERT#1]
Although the loss gets smaller when training, in testing model only predict one class for the entire test set.

The only exception was when I was training with my data (balanced binary classification data) after ossilating between the two classes through epochs (always achieving 50% on dev), in epoch 20 the model finally predicted different classes for different samples and achieved a score around 60% on development set. I could not replicate this phenomenon again.

**Comment by OsmanMutlu** [comment]
This issue is more or less solved with commit [1077784](https://github.com/OsmanMutlu/Pytorch-GANBERT/commit/10777848796ec46be048faac4deaf67f3b96ddb6).  Though in some runs it still appears to be an issue.
